python/cudf/cudf/core/index.py

```diff
@@ -914,17 +947,46 @@ def _from_table(cls, table):
 
 
 class RangeIndex(Index):
-    def __new__(cls, start, stop=None, name=None):
-        """An iterable integer index defined by a starting value and ending value.
-        Can be sliced and indexed arbitrarily without allocating memory for the
-        complete structure.
+    """
+    Immutable Index implementing a monotonic integer range.
+
+    This is the default index type used by DataFrame and Series
+    when no explicit index is provided by the user.
+
+    Parameters
+    ----------
+    start : int (default: 0), or other RangeIndex instance
```

Comment:
It doesn't look like we handle the situation where `start` is a `RangeIndex` currently

Reply:
It should be `range`, will update that